packages/intrepid2/src/Discretization/Basis/Intrepid2_DerivedBasis_HCURL_HEX.hpp

```diff
@@ -631,7 +627,7 @@ namespace Intrepid2
         \param [in] subCellOrd - position of the subCell among of the subCells having the same dimension
         \return pointer to the subCell basis of dimension subCellDim and position subCellOrd
      */
-    BasisPtr<ExecutionSpace, OutputValueType, PointValueType>
+    Teuchos::RCP<BasisBase>
```

Comment:
Is this change meaningful ? This means that the ``Teuchos::RCP<BasisBase>`` should be the same as ``BasisPtr``. That just means that the BasisBase should be defined and inherited from the Basis class. I am not sure how this improves things in your implementation.

Reply:
Yes, it is meaningful.  BasisBase may have as its first template argument a DeviceType that is distinct from the ExecutionSpace.  If so, then `BasisPtr<ExecutionSpace, OutputValueType, PointValueType>` does not match the return type in the base class.

Reply:
Does this work in the virtual override context if you change ?

Reply:
I'm not entirely sure what you are asking.  If you are asking whether things compile if I don't make this change, the answer is no, they do not compile.

Reply:
I mean that both returns teuchos rcp and the compiler may consider this class is overriding function. If they consider this function is different, the compiler may redirect the function to its super class interface. Previously, it probably does not compile because it uses execution space template argument instead of device argument. As you mentioned, it compiles but does it override the function correctly when the object is casted to its super class (in other words, does this make sure that it has the same function signature to override) ? Unit test will tell this.

Reply:
@kyungjoo-kim, the method is marked as override.  If the compiler considered this to be a different function from that in the base class, it would refuse to compile it.